csrc/ir/container.h

```diff
@@ -144,6 +153,14 @@ class IrContainer : public PolymorphicBase {
   void assumePositive(Val* val);
   void assumeNonNegative(Val* val);
 
+  // A simple garbage collection mechanism to remove all Exprs and Vals that
```

Comment:
This seems to be coupled with `StatementGuard`. Should we mention that here in the comment?

Reply:
Done -- made it private and added StatementGuard as a friend class